src/core/Microsoft.PowerToys.Settings.UI/Controls/HotkeySettingsControl.xaml.cs

```diff
@@ -135,15 +144,106 @@ private void KeyEventHandler(int key, bool matchValue, int matchValueCode, strin
+                {
+                    return false;
+                }
+
+                // Shift was pressed on entering but it was released and later pressed again.
+                // Ignore the tab key and the system already has the shift key pressed, therefore this would simulate Shift+Tab.
+                // However, since the last shift key was only used to move out of the control, reset the status of shift within the control.
+                else if (internalSettings.Shift && _shiftKeyDownOnEntering && _shiftToggled && !internalSettings.Win && !internalSettings.Alt && !internalSettings.Ctrl)
+                {
+                    internalSettings.Shift = false;
+
+                    return false;
+                }
+
+                // Shift was pressed on entering and was later released.
+                // The system still has shift in the key pressed status, therefore pass a Shift KeyUp message to the system, to release the shift key, therefore simulating only the Tab key press.
+                else if (!internalSettings.Shift && _shiftKeyDownOnEntering && _shiftToggled && !internalSettings.Win && !internalSettings.Alt && !internalSettings.Ctrl)
+                {
+                    NativeKeyboardHelper.INPUT inputShift = new NativeKeyboardHelper.INPUT
+                    {
+                        type = NativeKeyboardHelper.INPUTTYPE.INPUT_KEYBOARD,
+                        data = new NativeKeyboardHelper.InputUnion
+                        {
+                            ki = new NativeKeyboardHelper.KEYBDINPUT
+                            {
+                                wVk = 0x10,
+                                dwFlags = (uint)NativeKeyboardHelper.KeyEventF.KeyUp,
+                                dwExtraInfo = (UIntPtr)extraInfoIgnoreKeyEvent,
+                            },
+                        },
+                    };
+
+                    NativeKeyboardHelper.INPUT[] inputs = new NativeKeyboardHelper.INPUT[] { inputShift };
+
+                    _ = NativeMethods.SendInput(1, inputs, NativeKeyboardHelper.INPUT.Size);
+
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
         private async void Hotkey_KeyDown(int key)
         {
             await Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
             {
                 KeyEventHandler(key, true, key, Lib.Utilities.Helper.GetKeyName((uint)key));
                 if (internalSettings.Code > 0)
                 {
-                    lastValidSettings = internalSettings.Clone();
-                    HotkeyTextBox.Text = lastValidSettings.ToString();
+                    latestSettings = internalSettings.Clone();
+
+                    if (latestSettings != null && (latestSettings.IsValid() || latestSettings.IsEmpty()))
```

Comment:
I modified the code so that this piece of code is no longer there but to answer your question the `IsEmpty` case is needed after we press the escape key to clear the contents of the hotkey control. IsValid checks if atleast one modifier is pressed along with a non-modifier key. However, the empty case is also a valid scenario that we want to display to the user.

To add onto this without the changes in this PR, the way the hotkey control works is that any invalid setting (say just a letter as well) would show up in the hotkey while being typed. However, when the hotkey is about to lose focus, the lastvalidsetting config would be set. 

I've now modified the PR so that the behavior remains the same as before. It only does not show tab and shift+Tab when the user types it.